src/App.js

```diff
@@ -0,0 +1,198 @@
+
+  const displayAllBoards = () => {
+    getAllBoardsAsync()
+      .then((boards) => {
+        setBoardData(boards);
+      })
+      .catch((err) => {
+        console.log(err);
+        setErrorMessage("error displaying all boards");
+      });
+  };
+
+  useEffect(() => {
+    displayAllBoards();
+  }, [boardData]);
+
+  const selectBoard = (id) => {
+    selectBoardAsync(id)
+      .then((board) => {
+        setSelectedBoardId(board.boardId);
+      })
+      .catch((err) => {
+        console.log(err.message);
+        setErrorMessage("error displaying board");
+      });
+  };
+
+  const postBoard = (boardData) => {
+    postBoardAsync(boardData)
+      .then((newBoard) => {
+        setBoardData((oldData) => [...oldData, newBoard]);
+      })
+      .catch((err) => {
+        console.log(err.message);
+        setErrorMessage(err.message);
+      });
+  };
+
+  const postCard = (cardData) => {
+    postCardAsync(cardData, selectedBoardId)
+      .then(() => {
+        setBoardData((oldBoardData) => {
+          const newBoardData = oldBoardData.map((board) => {
+            if (board.boardId === selectedBoardId) {
+              return getSelectedBoard(selectedBoardId);
+            } else {
+              return board;
+            }
+          });
+          return newBoardData;
+        });
+      })
+      .catch((err) => {
+        console.log(err.message);
+        setErrorMessage(err.message);
+      });
+  };
+
+  const likeCard = (cardId) => {
+    likeCardAsync(cardId, selectedBoardId)
+      .then(() => {
+        setBoardData((oldBoardData) => {
+          const newBoardData = oldBoardData.map((board) => {
+            if (board.boardId === selectedBoardId) {
+              return getSelectedBoard(selectedBoardId);
+            } else {
+              return board;
+            }
+          });
+          return newBoardData;
+        });
+      })
+      .catch((err) => {
+        console.log(err.message);
+        setErrorMessage(err.message);
+      });
+  };
+
+  const deleteCard = (cardId) => {
+    deleteCardAsync(cardId, selectedBoardId)
+      .then(() => {
+        setBoardData((oldBoardData) => {
+          const newBoardData = oldBoardData.map((board) => {
+            if (board.boardId === selectedBoardId) {
+              return getSelectedBoard(selectedBoardId);
+            } else {
+              return board;
+            }
+          });
+          return newBoardData;
+        });
+      })
+      .catch((err) => {
+        console.log(err.message);
+        setErrorMessage(err.message);
+      });
+  };
+
+  const handleNewBoard = (formFields) => {
+    postBoard(formFields);
+  };
+
+  const handleNewCard = (formFields) => {
+    if (formFields.message.length <= 40) {
+      postCard(formFields);
+    } else {
+      setErrorMessage("message is too long");
+    }
+  };
+
+  const handleLike = (id) => {
+    likeCard(id);
+  };
+
+  const handleDelete = (id) => {
+    deleteCard(id);
+  };
```

Comment:
The comment for `handleNewBoard` applies to these functions as well: could `likeCard` and `deleteCard` be passed forward instead since `handleLike` and `handleDelete` don't contain other statements?